gen/generate.go

```diff
@@ -165,6 +169,38 @@ func Generate(m *compile.Module, o *Options) error {
 	return nil
 }
 
+func duplicateAfterNormalization(fileName string) error {
+	absFileName, err := filepath.Abs(fileName)
+	if err != nil {
+		return generateError{Name: fileName, Reason: fmt.Errorf("File %q does not exist: %v", fileName,
+			err)}
+	}
+	normalizedFileName := normalizeFilePath(absFileName)
+	if fileNormalized(absFileName, normalizedFileName) && fileExists(normalizedFileName) {
+		return generateError{
+			Name:   fileName,
+			Reason: fmt.Errorf("normalized file %q collides with existing file %q", normalizedFileName, absFileName)}
+	}
+	return nil
+}
+
+func fileNormalized(f1, f2 string) bool {
+	return f1 != f2
+}
+
+func fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
```

Comment:
let's add a small comment here since the functionality isn't immediately obvious.

```suggestion
// normalizeFilePath replaces hyphens in the file name with underscores.
```